web-console/src/views/load-data-view.tsx

```diff
@@ -0,0 +1,2353 @@
+              </Popover>
+            </FormGroup>
+            {
+              dimensionMode === 'auto-detect' &&
+              <AutoForm
+                fields={[
+                  {
+                    name: 'dataSchema.parser.parseSpec.dimensionsSpec.dimensionExclusions',
+                    label: 'Exclusions',
+                    type: 'string-array',
+                    info: <>
+                      Provide a comma separated list of columns (use the column name from the raw data) you do not want Druid to ingest.
+                    </>
+                  }
+                ]}
+                model={spec}
+                onChange={s => this.updateSpec(s)}
+              />
+            }
+            <FormGroup>
+              <Switch
+                checked={rollup}
+                onChange={() => this.setState({ newRollup: !rollup })}
+                labelElement="Rollup"
+              />
+              <Popover
+                content={
+                  <div className="label-info-text">
+                    <p>
+                      If you enable roll-up, Druid will try to pre-aggregate data before indexing it to conserve storage.
+                      The primary timestamp will be truncated to the specified query granularity, and rows containing the same string field values will be aggregated together.
+                    </p>
+                    <p>
+                      If you enable rollup, you must specify which columns are <a href="http://druid.io/docs/latest/ingestion/ingestion-spec.html#dimensionsspec">dimensions</a> (fields you want to group and filter on), and which are <a href="http://druid.io/docs/latest/querying/aggregations.html">metrics</a> (fields you want to aggregate on).
+                    </p>
+                  </div>
+                }
+                position="left-bottom"
+              >
+                <Icon icon={IconNames.INFO_SIGN} iconSize={14}/>
+              </Popover>
+            </FormGroup>
+            <AutoForm
+              fields={[
+                {
+                  name: 'dataSchema.granularitySpec.queryGranularity',
+                  label: 'Query granularity',
+                  type: 'string',
+                  suggestions: ['NONE', 'MINUTE', 'HOUR', 'DAY'],
+                  info: <>
+                    This granularity determines how timestamps will be truncated (not at all, to the minute, hour, day, etc).
+                    After data is rolled up, this granularity becomes the minimum granularity you can query data at.
+                  </>
+                }
+              ]}
+              model={spec}
+              onChange={s => this.updateSpec(s)}
+            />
+          </>
+        }
+        {!selectedMetricSpec && this.renderDimensionSpecControls()}
+        {!selectedDimensionSpec && this.renderMetricSpecControls()}
+        {this.renderChangeRollupAction()}
+        {this.renderChangeDime
```

Comment:
nit: same about reading a bit strange, but also copied from the docs...

Reply:
Yeah I was going to submit an issue about that part being a bit unclean in the druid ingestion spec.